test/native/setup.js

```diff
@@ -39,8 +34,12 @@ jest.mock( '@wordpress/element', () => {
 	};
 } );
 
-jest.mock( '@wordpress/api-fetch', () => jest.fn() );
-apiFetch.setFetchHandler = jest.fn();
+jest.mock( '@wordpress/api-fetch', () => {
+	const apiFetchMock = jest.fn();
+	apiFetchMock.setFetchHandler = jest.fn();
```

Comment:
API fetch module exports some of the functionality as properties of the function ([reference](https://github.com/WordPress/gutenberg/blob/43d950788629daaf588c80a2a56173124f112c60/packages/api-fetch/src/index.js#L188-L197)), for this reason, the mock is structured this way.